extensions-contrib/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleDataSegmentKiller.java

```diff
@@ -52,11 +52,9 @@ public void kill(DataSegment segment) throws SegmentLoadingException
     Map<String, Object> loadSpec = segment.getLoadSpec();
     final String bucket = MapUtils.getString(loadSpec, "bucket");
     final String indexPath = MapUtils.getString(loadSpec, "path");
-    final String descriptorPath = indexPath.substring(0, indexPath.lastIndexOf('/')) + "/descriptor.json";
 
     try {
       deleteIfPresent(bucket, indexPath);
-      deleteIfPresent(bucket, descriptorPath);
```

Comment:
I think we might still need to have the code to delete the descriptor files in SegmentKillers, so that any segments which are created pre 0.14 release when deleted using kill tasks are completely killed.
Can we keep the code to deal with deletion of descriptors and add a comment on why it is still kept ?

Reply:
Thanks. It sounds good. I reverted back and added a comment.